src/Service/Proxy.php

```diff
@@ -61,6 +61,23 @@ public function providerData(string $package, int $expireTime = self::PACKAGES_E
         return $this->metadataProvider->fromUrl($this->getUrl($providerPath->get()));
     }
 
+    /**
+     * @return Option<array<mixed>>
+     */
+    public function providerDataV2(string $package, int $expireTime = self::PACKAGES_EXPIRE_TIME): Option
+    {
+        if (!($fromPath = $this->metadataProvider->fromPath($package, $this->url, $expireTime))->isEmpty()) {
+            return $fromPath;
+        }
+
+        $providerPath = $this->getProviderPathV2($package);
+        if ($providerPath->isEmpty()) {
+            return Option::none();
+        }
+
+        return $this->metadataProvider->fromUrl($this->getUrl($providerPath->get()));
```

Comment:
We will probably need to do it differently, in this case we will overwrite the v1 data with the v2 version.
If I find a moment, I can finish it, but currently I am still involved in something else, if you would like to finish it yourself, I suggest you do it through an additional parameter and save the data in a separate directory `p2`